openstack/nova/templates/_helpers.tpl

```diff
@@ -76,8 +76,46 @@ annotations:
   {{- $name := index . 1 }}
   {{- with index . 0 }}
     {{- $bin := include (print .Template.BasePath "/bin/_" $name ".tpl") . }}
-    {{- $all := list $bin (include "utils.proxysql.job_pod_settings" . ) (include "utils.proxysql.volume_mount" . ) (include "utils.proxysql.container" . ) (include "utils.proxysql.volumes" .) | join "\n" }}
+    {{- $all := list $bin (include "utils.proxysql.job_pod_settings" . ) (include "utils.proxysql.volume_mount" . ) (include "utils.proxysql.container" . ) (include "utils.proxysql.volumes" .) (tuple . (dict) | include "utils.snippets.kubernetes_entrypoint_init_container") | join "\n" }}
     {{- $hash := empty .Values.proxysql.mode | ternary $bin $all | sha256sum }}
 {{- .Release.Name }}-{{ $name }}-{{ substr 0 4 $hash }}-{{ .Values.imageVersion | required "Please set nova.imageVersion or similar"}}
   {{- end }}
 {{- end }}
+
+
+{{- define "nova.helpers.database_services" }}
+  {{- $envAll := . }}
+  {{- $dbs := dict }}
+  {{- range $d := $envAll.Chart.Dependencies }}
+    {{- if and (hasPrefix "mariadb" $d.Name) }}
+        {{- $db := get $envAll.Values $d.Name }}
+        {{- if get $db "enabled" }}
+          {{- $_ := set $dbs (print (get $db "name") "-mariadb") $db }}
+        {{- end }}
+    {{- end }}
+  {{- end }}
+  {{- keys $dbs | sortAlpha | join "," }}
+{{- end }}
+
+
+{{- define "nova.helpers.cell01_services" }}
+  {{- print .Values.mariadb_api.name "-mariadb," .Values.mariadb.name "-mariadb," .Release.Name "-rabbitmq" }}
+{{- end }}
+
+{{- define "nova.helpers.cell1_services" }}
+  {{- print .Values.mariadb.name "-mariadb," .Release.Name "-rabbitmq" }}
+{{- end }}
+
+{{- define "nova.helpers.cell2_services" }}
+  {{- if .Values.cell2.enabled }}
+    {{- print .Values.mariadb_cell2.name "-mariadb," .Values.rabbitmq_cell2.name "-rabbitmq" }}
```

Comment:
I just took another look at how this renders, and it looks like we cannot use `.Values.rabbitmq_cell2.name`. It generates `nova-cell2-rabbitmq` instead of `nova-cell-vc-b-1-rabbitmq` in qa.

I checked all cell2 regions and `.Values.rabbitmq_cell2.nameOverride` would work i.e.
```diff
diff --git a/openstack/nova/templates/_helpers.tpl b/openstack/nova/templates/_helpers.tpl
index 2706d3861..40b817101 100644
--- a/openstack/nova/templates/_helpers.tpl
+++ b/openstack/nova/templates/_helpers.tpl
@@ -108,7 +108,7 @@ annotations:
 
 {{- define "nova.helpers.cell2_services" }}
   {{- if .Values.cell2.enabled }}
-    {{- print .Values.mariadb_cell2.name "-mariadb," .Values.rabbitmq_cell2.name "-rabbitmq" }}
+    {{- print .Values.mariadb_cell2.name "-mariadb," .Release.Name "-" .Values.rabbitmq_cell2.nameOverride }}
   {{- end }}
 {{- end }}
 
```